client/v3/balancer/balancer.go

```diff
@@ -191,6 +217,16 @@ func (bb *baseBalancer) HandleResolvedAddrs(addrs []resolver.Address, err error)
 			// (DO NOT) delete(bb.scToSt, sc)
 		}
 	}
+
+	// TODO: Consider just returning `ErrBadResolverState` if `warnedErrors` is
+	//       not empty. Or (to make a breaking change) just early exit with a
+	//       `ErrBadResolverState` when `warnedErrors` would first be appended to.
+	return multierr.Combine(warnedErrors...)
```

Comment:
@ptabor FYI I vetted that `multierr.Combine()` will return `nil` when `warnedErrors` is empty (and even in cases where every error in `warnedErrors` is `nil`)